tests/test_init.py

```diff
@@ -26,3 +26,208 @@ def test_systemd_present(container):
+    )
+    assert (
+        "0 loaded units listed" in output.stdout
+    ), "failed systemd units detected"
+
+
+@pytest.mark.parametrize("container", [INIT_CONTAINER], indirect=True)
+@pytest.mark.skipif(
+    DOCKER_SELECTED
+    and (int(LOCALHOST.package("systemd").version.split(".")[0]) >= 248),
+    reason="Running systemd in docker is broken as of systemd 248, see https://github.com/moby/moby/issues/42275",
+)
+def test_systemd_nofailed_units(container):
+    """
+    Check that there are not failed systemd units present
+    """
+    output = container.connection.run_expect(
+        [0], "systemctl list-units --state=failed"
+    )
+    assert (
+        "0 loaded units listed" in output.stdout
+    ), "failed systemd units detected"
+
+
+@pytest.mark.parametrize("container", [INIT_CONTAINER], indirect=True)
+@pytest.mark.skipif(
+    DOCKER_SELECTED
+    and (int(LOCALHOST.package("systemd").version.split(".")[0]) >= 248),
+    reason="Running systemd in docker is broken as of systemd 248, see https://github.com/moby/moby/issues/42275",
+)
+def test_systemd_detect_virt(container, host, container_runtime):
+    """
+    Check that systemd-detect-virt correctly detects the used container runtime
+    """
+    output = container.connection.run_expect([0], "systemd-detect-virt")
+    runtime = container_runtime.runner_binary
+    assert (
+        runtime in output.stdout
+    ), f"systemd-detect-virt failed to detect {runtime}"
+
+
+@pytest.mark.parametrize("container", [INIT_CONTAINER], indirect=True)
+@pytest.mark.skipif(
+    DOCKER_SELECTED
+    and (int(LOCALHOST.package("systemd").version.split(".")[0]) >= 248),
+    reason="Running systemd in docker is broken as of systemd 248, see https://github.com/moby/moby/issues/42275",
+)
+def test_journald(container):
+    """
+    Check that journald behaves correctly
+    """
+
+    # Check that we reached at least the multiuser target
+    journal = container.connection.run_expect([0], "journalctl --boot -e")
+    assert (
+        "Reached target Multi-User System" in journal.stdout
+    ), f"Multi-User target was not reached"
+
+
+@pytest.mark.parametrize("container", [INIT_CONTAINER], indirect=True)
+@pytest.mark.skipif(
+    DOCKER_SELECTED
+    and (int(LOCALHOST.package("systemd").version.split(".")[0]) >= 248),
+    reason="Running systemd in docker is broken as of systemd 248, see https://github.com/moby/moby/issues/42275",
+)
+def test_hostnamectl(container, host, container_runtime):
+    """
+    Check that hostnamectl prints the expected values
+    """
+
+    # Process the given output into a key:value map, one entry by line
+    def split_values(output, sep=":"):
+        ret = {}
+        for line in output.split("\n"):
+            line = line.strip()
+            i = line.find(sep)
+            if i < 0:
+                continue
+            name, value = line[:i].strip(), line[i + 1 :].strip()
+            ret[name] = value
```

Comment:
How about something like this?
```suggestion
            tmp = line.strip().split(sep)
            if len(tmp) != 2:
                continue
            name, value = tmp
            ret[name] = value
```